src/datachain/query/dataset.py

```diff
@@ -1097,26 +1091,43 @@ def __init__(
         self.temp_table_names: list[str] = []
         self.dependencies: set[DatasetDependencyType] = set()
         self.table = self.get_table()
-        self.starting_step: StartingStep
+        self.query_step: Optional[QueryStep] = None
         self.name: Optional[str] = None
         self.version: Optional[int] = None
         self.feature_schema: Optional[dict] = None
         self.column_types: Optional[dict[str, Any]] = None
 
-        self.name = name
+        # for listing pre-step
+        self.list_fn: Optional[Callable] = None
+        self.list_ds_name: Optional[str] = None
 
-        if fallback_to_studio and is_token_set():
-            ds = self.catalog.get_dataset_with_remote_fallback(name, version)
+        self.name = name
+        self.dialect = self.catalog.warehouse.db.dialect
+        if version:
+            self.version = version
+
+        if is_listing_dataset(self.name):
```

Comment:
Even if they are the same (`self.name` and `name`), we are using `name` below in this branch, so IMO it should be either `name` or `self.name` everywhere, just to be consistent.

```suggestion
        if is_listing_dataset(name):
```